src/components/forms/shipment/ShipmentForm.tsx

```diff
@@ -0,0 +1,190 @@
+      payload.id = initialFormValue.id;
+      delete payload.order_ids;
+      delete payload.order_names;
+      delete payload.orders;
+    }
+    return payload;
+  };
+
+  const onSubmitAdd = async (formValues: FieldValues) => {
+    const customers = Array.from(new Set(orderDetails.map((row: any) => (row.customer))));
+    if (customers.length > 1) {
+      customers.forEach(async (customerId, index) => {
+        const orders = orderDetails
+          .filter((order: any) => order.customer === customerId)
+          .map((row: any) => (row.order_details_id));
+        const payload = formatShipmentPayload(formValues, orders);
+        await createMultipleShipments(payload);
+        if (index === customers.length - 1) {
+          navigate(`${Paths.shipment_details}`);
+        }
+      });
+    } else {
+      const orders = orderDetails.map((row: any) => (row.order_details_id));
+      const payload = formatShipmentPayload(formValues, orders);
+      dispatch(postShipmentRequest(payload as PostShipmentRequestPayload));
+      navigate(`${Paths.shipment_details}`);
+    }
+  };
+
+  const onSubmitEdit = async (formValues: FieldValues) => {
+    const orders = orderDetails.map((row: any) => (row.order_details_id));
+    const payload = formatShipmentPayload(formValues, orders);
+    await updateShipment(payload);
+    await updateShipmentOrders(payload.id, orders);
+    navigate(`${Paths.shipment_details}`);
+  };
+
+  const onSubmit = async (formValues: FieldValues) => {
+    if (mode === 'Add') {
+      onSubmitAdd(formValues);
+    } else {
+      onSubmitEdit(formValues);
+    }
+  };
+
+  return (
+    <div className="bg-transit-white w-full rounded-lg pt-8 px-4" style={{ width: '75%', margin: '0 auto' }}>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div className="flex flex-col gap-4">
+          <FormHeader title={title} onClick={onCancel} />
+          <ShipmentBaseData
+            control={control}
+            register={register}
+            errors={errors}
+            watch={watch}
+            setValue={setValue}
+            mode={mode}
+            initialFormValue={initialFormValue}
+          />
+          <ShipmentOrderData
+            control={control}
+            register={register}
+            errors={errors}
+            watch={watch}
+            setValue={setValue}
+            onOrderDetailAdd={onOrderDetailAdd}
+            onOrderDetailDelete={onOrderDetailDelete}
+            orderDetails={orderDetails}
+            mode={mode}
+          />
+          <p className="float-left text-[21px] text-transit-black font-semibold">Delivery Details</p>
+          <ShipmentDetailsData
+            control={control}
+            register={register}
+            errors={errors}
+            watch={watch}
+            setValue={setValue}
+            mode={mode}
+            initialFormValue={initialFormValue}
+          />
+          <p className="float-left text-[21px] text-transit-black font-semibold" />
```

Comment:
This p is empty - why is here?

Reply:
I think this p give us empty line.

Reply:
If you need empty line add padding above or add gap ;)